File: backend/src/main/java/com/woon7713/backend/entity/Comment.java
package com.woon7713.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "comments",
        indexes = {
                @Index(name = "idx_comment_post_id", columnList = "post_id"),
                @Index(name = "idx_comment_user_id", columnList = "user_id"),
                @Index(name = "idx_comment_post_created", columnList = "post_id, created_at"),
        }
)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Comment {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;

        @Column(columnDefinition = "TEXT", nullable = false)
        private String content;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "user_id", nullable = false)
        private User user;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "post_id", nullable = false)
        private Post post;

        @CreationTimestamp
        @Column(name = "created_at", updatable = false)
        private LocalDateTime createdAt;
}

File: frontend/src/services/comment.js
import api from "./api";

const commentService = {
  createComment: async (postId, content) => {
    const response = await api.post(`/api/comments/posts/${postId}`, {
      content,
    });
    return response.data;
  },

  fetchComments: async (postId) => {
    const response = await api.get(`/api/comments/posts/${postId}`);
    return response.data.content;
  },
};

export default commentService;
